web/index.html

```diff
@@ -78,20 +80,18 @@
 	// reset zoom
 	document.body.style.zoom = (window.innerWidth / window.outerWidth);
 
-	const canvas = document.getElementById('canvas');
+	var canvas = document.getElementById('canvas');
+	var width = window.outerWidth;
+	var height = window.outerHeight;
 	if (mobileAndTabletCheck()) {
-		canvas.width = window.innerWidth;
-		canvas.height = window.innerHeight;
-	}
-	else {
-		canvas.width = window.outerWidth;
-		canvas.height = window.outerHeight;
+		width = window.innerWidth;
+		height = window.innerHeight;
 	}
 
 	var Module = {};
 	Module.canvas = canvas;
 	Module.onRuntimeInitialized = () => {
-		Main.run(canvas);
+		Main.run(canvas, width, height);
```

Comment:
Only setting the canvas width and height once (once we've actually bounced off the wasm code) is slightly less weird. I still need a loader.